src/Avatar/Avatar.js

```diff
@@ -0,0 +1,181 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+import React, { Children, cloneElement } from 'react';
+import { createStyledComponent, getNormalizedValue, pxToEm } from '../styles';
+import colors from '../colors';
+
+type Props = {
+  /** When `children` is a string, this will be the rendered text */
+  abbr?: string,
+  /**
+   * `img` (with an `alt` attribute), [Icon](../icon) (with a `title`), or a
+   * string
+   */
+  children: React$Node,
+  /** Background color */
+  color?:
+    | 'blue'
+    | 'dusk'
+    | 'gray'
+    | 'green'
+    | 'indigo'
+    | 'lime'
+    | 'magenta'
+    | 'orange'
+    | 'purple'
+    | 'red'
+    | 'sky'
+    | 'slate'
+    | 'teal'
+    | 'theme'
+    | 'yellow',
+  /** Available sizes */
+  size?: 'small' | 'medium' | 'large' | 'jumbo'
+};
+
+const textColor = {
+  blue: 'white',
+  dusk: 'white',
+  gray: 'black',
+  green: 'black',
+  indigo: 'white',
+  lime: 'black',
+  magenta: 'white',
+  orange: 'white',
+  purple: 'white',
+  red: 'white',
+  sky: 'black',
+  slate: 'white',
+  teal: 'black',
+  theme: 'white',
+  yellow: 'black'
+};
+
+const iconSize = {
+  small: 'medium',
+  medium: 'medium',
+  large: pxToEm(24),
+  jumbo: pxToEm(24)
+};
+
+export const componentTheme = (baseTheme: Object) => ({
+  Avatar_fontSize_small: baseTheme.fontSize_mouse,
+  Avatar_fontSize_medium: baseTheme.fontSize_ui,
+  Avatar_fontSize_large: baseTheme.fontSize_h4,
+  Avatar_fontSize_jumbo: baseTheme.fontSize_h4,
+  Avatar_fontWeight: baseTheme.fontWeight_bold,
+  Avatar_size_small: baseTheme.size_small,
+  Avatar_size_medium: baseTheme.size_medium,
+  Avatar_size_large: baseTheme.size_large,
+  Avatar_size_jumbo: baseTheme.size_jumbo,
+
+  ...baseTheme
+});
+
+const Root = createStyledComponent(
+  'span',
+  props => {
+    const theme = componentTheme(props.theme);
+
+    const color = theme[`color_${textColor[props.color]}`];
+    const fontSize = !props.icon
+      ? theme[`Avatar_fontSize_${props.size}`]
+      : '1em';
+    const size = getNormalizedValue(
+      theme[`Avatar_size_${props.size}`],
+      fontSize
+    );
+
+    return {
+      alignItems: 'center',
+      backgroundColor:
+        props.color === 'theme'
+          ? theme.color_theme_60
+          : colors[`${props.color}_60`],
+      color,
+      borderRadius: '100%',
+      display: 'inline-flex',
+      fontSize,
+      fontWeight: theme.Avatar_fontWeight,
+      height: size,
+      lineHeight: size,
+      justifyContent: 'center',
+      verticalAlign: 'middle',
+      width: size,
+
+      '& > abbr': {
+        textDecoration: 'none'
+      },
+
+      '& > img': {
+        borderRadius: '100%',
+        display: 'block',
+        width: '100%'
+      },
+
+      '& > [role="img"]': {
+        fill: color
+      }
+    };
+  },
+  {
+    displayName: 'Avatar',
+    includeStyleReset: true,
+    filterProps: ['icon']
```

Comment:
Is `filterProps` still needed if you are only using `children` now?

Reply:
Nope.